src/android/CameraActivity.java

```diff
@@ -355,7 +355,15 @@ public void onPreviewFrame(final byte[] data, final Camera camera) {
             public void run() {
 
               //raw picture
-              byte[] bytes = mPreview.getFramePicture(data, camera, width, height, quality); // raw bytes from preview
+              int w = width;
+              int h = height;
+              if (w == 0 || h == 0) {
+                // provide default width / height
+                Camera.Parameters params = camera.getParameters();
+                w = params.getPreviewSize().width;
+                h = params.getPreviewSize().height;
```

Comment:
I am fairly certain this is what was causing #182 . I made my own set of changes in the latest commit which follows your PR. It has a different method of getting the best width and height if they are 0.